packages/backend/src/models/GroupsModel.ts

```diff
@@ -187,38 +189,83 @@ export class GroupsModel {
-            RETURNING groups.group_uuid, groups.name, groups.created_at, organizations.organization_uuid
-            `,
-            {
-                groupUuid,
-                name: update.name,
-            },
-        );
-        const [updatedGroup] = results.rows;
-        if (updatedGroup === undefined) {
+    async updateGroup(
+        groupUuid: string,
+        update: UpdateGroupWithMembers,
+    ): Promise<Group | GroupWithMembers> {
+        // TODO: fix include member count
+        const existingGroup = await this.getGroupWithMembers(groupUuid, 10000);
+        if (existingGroup === undefined) {
             throw new NotFoundError(`No group found`);
         }
-        return {
-            uuid: updatedGroup.group_uuid,
-            name: updatedGroup.name,
-            createdAt: updatedGroup.created_at,
-            organizationUuid: updatedGroup.organization_uuid,
-        };
+
+        await this.database.transaction(async (trx) => {
+            let group;
+            if (update.name) {
+                group = await trx('groups')
+                    .update({ name: update.name })
+                    .where('group_uuid', groupUuid)
+                    .returning('*');
+            }
+            if (update.members !== undefined) {
+                const membersToAdd = differenceBy(
+                    update.members,
+                    existingGroup.members,
+                    'userUuid',
+                );
+                const membersToRemove = differenceBy(
+                    existingGroup.members,
+                    update.members || [],
+                    'userUuid',
+                );
+                if (membersToAdd.length > 0) {
+                    const newMembers = await trx
+                        .select([
+                            'groups.group_uuid',
+                            'users.user_id',
+                            'organization_memberships.organization_id',
+                        ])
+                        .from('groups')
+                        .innerJoin(
+                            'organization_memberships',
+                            'groups.organization_id',
+                            'organization_memberships.organization_id',
+                        )
+                        .innerJoin(
+                            'users',
+                            'organization_memberships.user_id',
+                            'users.user_id',
+                        )
+                        .whereIn(
+                            'users.user_uuid',
+                            membersToAdd.map((m) => m.userUuid),
+                        )
+                        .andWhere('groups.group_uuid', groupUuid);
+
+                    await trx('group_memberships')
+                        .insert(newMembers)
+                        .onConflict()
+                        .ignore()
+                        .returning('*');
```

Comment:
Interesting, we haven't used ignore before, what was the thought process?
We don't need the `.returning('*')` btw

Reply:
The thought process was to copy the behavior of the existing raw query for adding one group member which contains a `ON CONFLICT DO NOTHING`. So we'd be ignoring attempts to add the same member twice instead of throwing. Which seems fine to me, but if there is a reason to throw instead we can to that.

Reply:
makes sense 👍